src/Compilers/Core/Portable/CodeGen/PrivateImplementationDetails.cs

```diff
@@ -31,14 +31,37 @@ internal sealed class PrivateImplementationDetails : DefaultTypeDef, Cci.INamesp
         internal const string SynthesizedThrowIfNullFunctionName = "ThrowIfNull";
         internal const string SynthesizedThrowFunctionName = "Throw";
 
+        internal const string SynthesizedThrowSwitchExpressionExceptionFunctionName = "ThrowSwitchExpressionException";
+        internal const string SynthesizedThrowSwitchExpressionExceptionParameterlessFunctionName = "ThrowSwitchExpressionExceptionParameterless";
+        internal const string SynthesizedThrowInvalidOperationExceptionFunctionName = "ThrowInvalidOperationException";
+
         private readonly CommonPEModuleBuilder _moduleBuilder;       //the module builder
-        private readonly Cci.ITypeReference _systemObject;           //base type
-        private readonly Cci.ITypeReference _systemValueType;        //base for nested structs
+        private readonly SyntaxNode _syntaxNodeOpt;
+        private readonly DiagnosticBag _diagnostics;
+        private Cci.ITypeReference SystemObject //base type
```

Comment:
Let's simplify by reverting this to the original implementation as much as possible (fetching the needed special types eagerly in the constructor, etc.), as well as related changes in `src/Compilers/Core/Portable/Emit/CommonPEModuleBuilder.cs`.